---
title: "Report myelosuppresseion"
author: "Christelle Colin-Leitzinger"
date: "6/30/2020"
output: powerpoint_presentation
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.width = 10, fig.height = 7)
```
<br>

# Report
***

## Data
```{r}
# Import library
library(tidyverse)
library(data.table)
library(ggplot2)
library(viridis)
library(gtsummary)
library(VennDiagram)
library(ggpubr)
library(survival)

#######################################################################################  I  ### Load data----
path <- fs::path("","Volumes","Gillis_Research","Christelle Colin-Leitzinger", "CHIP_myelosuppression_M4M 2018")
#---
clinical <-
  read_csv(paste0(path, "/data/CHIP_CRASH_data_for_stats_v06.csv")) %>% 
  rename(old_CHIP = CHIP, old_CHPD = CHPD) %>% 
  select(-X32)

CHIP_muts1 <- 
  read_csv(paste0(path, "/data/cleaned CH_myelosupp_muts_2020_CC.csv"))
CHIP_muts <- 
  read.delim(paste0(path, "/data/M4M2018_updated 06.24.20_filtered.txt"))

#######################################################################################  II  ### Data cleaning----
# 2.1.Clinical data----
clinical <- clinical %>% 
  mutate(Cases_Controls = Case_Control) %>% 
  mutate(Case_Control = factor(Case_Control, labels = c("Cases", "Controls"), levels= c(1, 0))) %>% # 0 = ctrl
  # mutate(Case_Control = factor(Case_Control, labels= c("Controls", "Cases"))) %>% 
  mutate(old_CHIP = factor(old_CHIP, labels=c("No CHIP", "CHIP"))) %>% 
  mutate(old_CHPD= factor(old_CHPD, labels=c("No CHPD", "CHPD"))) %>% 
  mutate(Race = factor(Race, labels=c("White", "Other"))) %>% 
  mutate(Gender = factor(Gender, labels = c("Male", "Female"))) %>% 
  mutate(Smoking = factor(Smoking, labels=c("Never","Ever"))) %>% 
  mutate(mets = Mets) %>% 
  mutate(Mets = factor(Mets, labels=c("No", "Yes"))) %>%
  mutate(neutro = Neutro) %>% 
  mutate(Neutro = factor(Neutro, labels=c("No", "Yes"))) %>%
  mutate(anemia = Anemia) %>% 
  mutate(Anemia = factor(Anemia, labels=c("No", "Yes"))) %>%
  mutate(thrombo = Thrombo) %>% 
  mutate(Thrombo = factor(Thrombo, labels=c("No", "Yes"))) %>%
  mutate(leuko = Leuko) %>% 
  mutate(Leuko = factor(Leuko, labels=c("No", "Yes"))) %>%
  mutate(Prior_chemo = factor(Prior_chemo, labels=c("No", "Yes"))) %>% 
  mutate(Prior_rad = factor(Prior_rad, labels=c("No", "Yes"))) %>%
  filter(Cohort == "M4M") # remove later

# 2.2.CHIP data----
CHIP_muts <- full_join(CHIP_muts1 %>% 
                         select(patient_id),
                       CHIP_muts, 
                       by = "patient_id")

# Modified CHIP var to fit with the new data
CHIP_muts <- CHIP_muts %>% 
  mutate(CHIP = case_when(
    is.na(DATA) ~ 0,
    !is.na(DATA) ~ 1
  )) %>% 
  mutate(CHIP = factor(CHIP, labels= c("CHIP", "No CHIP"), levels= c(1, 0)))

#######################################################################################  III  ### Data binding----
# Will bind the 2 data but in 2 different ways 

# 3.1.Bind to have 1 mutation per row, multiple row per patient----
muts_data <- full_join(clinical[3:37], CHIP_muts, 
                          by = c("NGS_ID" = "patient_id")) %>%
  filter(str_detect(NGS_ID, "M4M")) %>% # remove later
  mutate(CHIP_VAF = case_when(
    VAF >= 0.1 ~ VAF
  )) %>% 
  select("NGS_ID", "Case_Control", "Strata", "old_CHIP", "CHIP", everything())
# write_csv(muts_data, paste0(path, "/Output/data output/muts_data.csv"))

# 3.2.Bind to have 1 patient per row, multiple mutation per row----
CHIP_muts1 <- dcast(setDT(CHIP_muts), patient_id ~ rowid(patient_id),
                    value.var = c("GENE", "FUNCTION", "COSMIC", "ESP6500", "VAF", "DEPTH", "CHIP")) %>% 
  select(-CHIP_2, -CHIP_3, -CHIP_4) %>% 
  rename(CHIP = CHIP_1)

global_data <- left_join(clinical[, c(1,3:37)], CHIP_muts1, 
                       by = c("NGS_ID" = "patient_id")) %>% 
  filter(str_detect(NGS_ID, "M4M")) %>% # remove later
  select("Patient", "NGS_ID", "Case_Control", "Strata", "old_CHIP", "CHIP", everything())

# write_csv(global_data, paste0(path, "/Output/data output/global_data.csv"))

rm(CHIP_muts1)
```
The data uses the cohort M4M completed with patient in UNC. Here we focused on M4M data only.
We have clinical information for `r lenght(clinical$NGS_ID)`.  
We also have mutation information aka `r lenght(unique(muts_data$NGS_ID))` on `r lenght(unique(CHIP_muts1$patient_id))` unique patient.  
We are going to investigate the presence or absence of CHIP/CHPD in patient treated with chemotherapie and showing a myelosuppression.
```{r, fig.width = 10, fig.height = 10}
p <- venn.diagram(
  x = list(Cases = c(unique(global_data[global_data$Case_Control == "Cases",]$NGS_ID)),
           Controls = c(unique(global_data[global_data$Case_Control == "Controls",]$NGS_ID)),
           CHIP = c(unique(global_data[global_data$CHIP == "CHIP",]$NGS_ID))),
  category.names = c("Cases", "Controls", "CHIP"),
  filename = NULL,
  # Output features
  imagetype="png" ,
  height = 900 ,
  width = 700 ,
  resolution = 300,
  compression = "lzw",

  # Circles
  main = "Mutations presented by cases_control",
  main.cex = 5,
  main.pos = c(0.5, .9),
  lwd = 2,
  lty = 'blank',
  fill = cividis(n=3),
  margin = 0.15,

  # Numbers
  cex = 5,
  fontface = "bold",
  fontfamily = "sans",
  # cat.pos = c(0, 180, 0),
  # cat.dist = c(0.055, -0.07, 0.015),
  cat.cex = 7
  #ext.percent = 2,

)
p
# jpeg("venn_jpeg.jpg");
grid.draw(p)
# dev.off()
```
## Clinical Data
```{r}
tbl <- as.data.frame(table(global_data$CANCER))
colourCount = length(unique(tbl$Var1))
getPalette = colorRampPalette(RColorBrewer::brewer.pal(8, "Accent"))(colourCount)
tbl %>% mutate(Var1 = fct_reorder(Var1, desc(Freq))) %>%
  ggplot(aes(x="", y=Freq, fill=Var1)) +
  geom_bar(stat="identity", width=1) +
  scale_fill_manual(name = "Cancer type", values = getPalette) +
  theme_minimal() +
  coord_polar("y", start=0, direction=-1) +
  labs(x="", y="", title="Cancer type Repartition")

tbl <- as.data.frame(table(global_data$CHIP,global_data$CANCER))
tbl %>% mutate(Var2 = fct_reorder(Var2, Freq)) %>%
  ggplot(aes(x=Var2, y=Freq, fill=Var1)) +
  geom_bar(stat="identity", width = .75, position = position_dodge(preserve = "single")) +
  scale_fill_discrete(name = "CHIP") +
  geom_text(aes(y = Freq + .5 , label=Freq), position = position_dodge(width = 0.9)) +
  theme_minimal() +
  coord_flip()+
  labs(x="Cancer Type", y="Number of Patient", title="CHIP in Cancer type Repartition")
```
```{r}
ggplot(data = global_data, aes(x=Gender, y=Age), fill=Gender) +
  geom_boxplot() +
  theme_minimal() +
  labs(x="Gender", y="Age", title="Age repartition") +
  geom_jitter(shape=16, position=position_jitter(0.2))+
  facet_grid(. ~ CHIP)
```
# Smoker
```{r}

```
## CHIP prevalence
***

```{r}
global_data %>% group_by(Case_Control,CHIP) %>% 
  summarise(count=n()) %>% 
  mutate(perc=(count/sum(count)*100)
  ) %>% 
  ggplot(aes(x=Case_Control, y= perc, fill=CHIP))+
  geom_bar(stat="identity") +
  labs(x = "", y = "percent", title = "Prevalence of CHIP") +
  theme_minimal()+
  geom_text(aes(label = round(perc,2)), size = 3, position = position_stack(vjust = 0.5))+
  geom_text(aes(label = paste0("n=", count)), size = 3, position = position_stack(vjust = 0.25))+
  annotate("text", x = 0, y = 105, label = paste0("p=",chisq.test(table(global_data$Case_Control, global_data$CHIP))[3]),
           color = "black", size = 6, hjust = 0, vjust = 1)
global_data %>% 
  select(Case_Control, CHIP) %>% 
  tbl_summary(by= Case_Control) %>% 
  add_p() %>% add_overall() %>% as_kable()
```


## Mutations-Variant Allele Frequencies


