---
title: "Report"
author: "Christelle Colin-Leitzinger"
date: "7/14/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      cache = FALSE,
                      fig.width = 6, fig.height = 5)
```
<br>

# Report
***

# Data
```{r}
# Import library
library(tidyverse)
library(data.table)
library(ggplot2)
library(viridis)
library(gtsummary)
library(VennDiagram)
library(ggpubr)
library(survival)

#######################################################################################  I  ### Load data----
path <- fs::path("","Volumes","Gillis_Research","Christelle Colin-Leitzinger", "CHIP_myelosuppression_M4M 2018")
#---
clinical <-
  read_csv(paste0(path, "/data/CHIP_CRASH_data_for_stats_v06.csv")) %>% 
  rename(old_CHIP = CHIP, old_CHPD = CHPD) %>% 
  select(-X32)

CHIP_muts1 <- 
  read_csv(paste0(path, "/data/cleaned CH_myelosupp_muts_2020_CC.csv"))
CHIP_muts <- 
  read.delim(paste0(path, "/data/M4M2018_updated 06.24.20_filtered.txt"))

#######################################################################################  II  ### Data cleaning----
# 2.1.Clinical data----
clinical <- clinical %>% 
  mutate(C_C = Case_Control) %>% 
  mutate(Case_Control = factor(Case_Control, labels = c("Cases", "Controls"), levels= c(1, 0))) %>% # 0 = ctrl
  # mutate(Case_Control = factor(Case_Control, labels= c("Controls", "Cases"))) %>% 
  mutate(old_CHIP = factor(old_CHIP, labels=c("No CHIP", "CHIP"))) %>% 
  mutate(old_CHPD= factor(old_CHPD, labels=c("No CHPD", "CHPD"))) %>% 
  mutate(Race = factor(Race, labels=c("White", "Other"))) %>% 
  mutate(Gender = factor(Gender, labels = c("Male", "Female"))) %>% 
  mutate(Smoking = factor(Smoking, labels=c("Never","Ever"))) %>% 
  mutate(Mets = factor(Mets, labels=c("No", "Yes"))) %>% 
  mutate(Neutro = factor(Neutro, labels=c("No", "Yes"))) %>% 
  mutate(Anemia = factor(Anemia, labels=c("No", "Yes"))) %>% 
  mutate(Thrombo = factor(Thrombo, labels=c("No", "Yes"))) %>% 
  mutate(Prior_chemo = factor(Prior_chemo, labels=c("No", "Yes"))) %>% 
  mutate(Prior_rad = factor(Prior_rad, labels=c("No", "Yes"))) %>%
  filter(Cohort == "M4M") # remove later

# 2.2.CHIP data----
CHIP_muts <- full_join(CHIP_muts1 %>% 
                         select(patient_id),
                       CHIP_muts, 
                       by = "patient_id")

# Modified CHIP var to fit with the new data
CHIP_muts <- CHIP_muts %>% 
  mutate(CHIP = case_when(
    is.na(DATA) ~ 0,
    !is.na(DATA) ~ 1
  )) %>% 
  mutate(CHIP = factor(CHIP, labels= c("CHIP", "No CHIP"), levels= c(1, 0)))

#######################################################################################  III  ### Data binding----
# Will bind the 2 data but in 2 different ways 

# 3.1.Bind to have 1 mutation per row, multiple row per patient----
muts_data <- full_join(clinical[3:32], CHIP_muts, 
                          by = c("NGS_ID" = "patient_id")) %>%
  filter(str_detect(NGS_ID, "M4M")) %>% # remove later
  select("NGS_ID", "Case_Control", "Strata", "old_CHIP", "CHIP", everything())

# 3.2.Bind to have 1 patient per row, multiple mutation per row----
CHIP_muts1 <- dcast(setDT(CHIP_muts), patient_id ~ rowid(patient_id),
                    value.var = c("GENE", "FUNCTION", "COSMIC", "ESP6500", "VAF", "DEPTH", "CHIP")) %>% 
  select(-CHIP_2, -CHIP_3, -CHIP_4) %>% 
  rename(CHIP = CHIP_1)

global_data <- left_join(clinical[, c(1,3:32)], CHIP_muts1, 
                       by = c("NGS_ID" = "patient_id")) %>% 
  filter(str_detect(NGS_ID, "M4M")) %>% # remove later
  select("Patient", "NGS_ID", "Case_Control", "Strata", "old_CHIP", "CHIP", everything())
```

The data uses the cohort M4M completed with patient in UNC. We have clinical information for `r length(clinical$NGS_ID)` patients.  
We also have mutation information aka `r length(unique(muts_data$NGS_ID))` on `r length(unique(CHIP_muts1$patient_id))` unique patient.  
The global data have `r dim(global_data)[2]` variables on `r dim(global_data)[1]` patients.  
`r sum(str_detect(global_data$Case_Control,"Cases"))` patients are cases of myelosuppression., `r sum(str_detect(global_data$Case_Control,"Controls"))` patients are controls.
We are going to investigate if the observed myelosuppression after chemotherapie is related to the presence or absence of CHIP in patient.
```{r}
rm(CHIP_muts1)
```


## CHIP prevalence

```{r, fig.width = 6, fig.height = 4}
global_data %>% group_by(Case_Control,CHIP) %>% 
  summarise(count=n()) %>% 
  mutate(perc=(count/sum(count)*100)
  ) %>% 
  ggplot(aes(x=Case_Control, y= perc, fill=CHIP))+
  geom_bar(stat="identity") +
  labs(x = "", y = "percent", title = "Prevalence of CHIP") +
  theme_minimal()+
  geom_text(aes(label = round(perc,2)), size = 3, position = position_stack(vjust = 0.5))+
  geom_text(aes(label = paste0("n=", count)), size = 3, position = position_stack(vjust = 0.25))+
  annotate("text", x = 0, y = 105, label = paste0("p=",chisq.test(table(global_data$Case_Control, global_data$CHIP))[3]),
           color = "black", size = 6, hjust = 0, vjust = 1)
global_data %>% 
  select(Case_Control, CHIP) %>% 
  tbl_summary(by= Case_Control) %>% 
  add_p() %>% add_overall()
```
```{r}
venn.diagram(
  x = list(Cases = c(unique(global_data[global_data$Case_Control == "Cases",]$NGS_ID)),
           Controls = c(unique(global_data[global_data$Case_Control == "Controls",]$NGS_ID)),
           CHIP = c(unique(global_data[global_data$CHIP == "CHIP",]$NGS_ID))),
  category.names = c("Cases", "Controls", "CHIP"),
  filename = NULL,
  output=FALSE,

  # Output features
  imagetype="png" ,
  height = 700 ,
  width = 700 ,
  resolution = 300,
  compression = "lzw",

  # Circles
  main = "Mutations presented by cases_control",
  main.cex = .5,
  main.pos = c(0.5, .9),
  lwd = 2,
  lty = 'blank',
  fill = cividis(n=3),
  margin = 0.15,

  # Numbers
  cex = .5,
  fontface = "bold",
  fontfamily = "sans",
  # cat.pos = c(0, 180, 0),
  # cat.dist = c(0.055, -0.07, 0.015),
  cat.cex = .7
  #ext.percent = 2,

)
```

## Mutations-Variant Allele Frequencies

```{r}
muts_data.summary <- muts_data %>%
  group_by(Case_Control) %>% 
  filter(!is.na(VAF)) %>% 
  summarise(
    sd = sd(VAF, na.rm = TRUE),
    VAF = mean(VAF, na.rm = TRUE),
    median = median(VAF, na.rm = TRUE),
    min = min(VAF, na.rm = TRUE),
    max = max(VAF, na.rm = TRUE),
    mean = mean(VAF, na.rm = TRUE)
  )
ggplot(muts_data %>% filter(!is.na(Case_Control)), aes(x=Case_Control, y= VAF, color=Case_Control)) +
  geom_jitter(
    position = position_jitter(0.2), color = "lightgray"
  ) + 
  theme_minimal()+
  geom_pointrange(
    aes(ymin = median-sd, ymax = median+sd),
    data = muts_data.summary
  )+
  stat_compare_means(label.x = 1.16, label.y = .4)
#
muts_data %>% 
  group_by(Case_Control) %>% 
  filter(!is.na(VAF)) %>% 
  select(Case_Control, VAF) %>% 
  tbl_summary(by= Case_Control, statistic = all_continuous() ~ "{median} ({sd})") %>% 
  add_p() %>% add_overall() %>% as_gt()
```

## Genes mutated in CHIP

```{r}
muts_data %>% 
  filter(!is.na(GENE)) %>% group_by(CHIP,GENE) %>% 
  summarise(count=n()) %>% 
  mutate(percent=(count/sum(count)*100)
  ) %>% 
  ggplot(aes(x=reorder(GENE, -percent), y=percent, fill = CHIP)) +
  geom_bar(stat = "identity", position = position_dodge2(preserve = "single")) + 
  coord_flip()

muts_data %>% filter(!is.na(GENE)) %>% group_by(Case_Control,GENE) %>% 
  summarise(count=n()) %>% 
  mutate(percent=(count/sum(count)*100))

muts_data %>% filter(!is.na(GENE)) %>% group_by(Case_Control,GENE) %>% 
  summarise(count=n()) %>% 
  mutate(percent=(count/sum(count)*100)
  ) %>% 
  ggplot(aes(x=reorder(GENE, percent), y=percent, fill = Case_Control)) +
  geom_bar(stat = "identity", position = position_dodge2(preserve = "single")) + 
  coord_flip()

muts_data %>% filter(!is.na(Case_Control)) %>% 
  select(Case_Control, GENE) %>% 
  tbl_summary(by=Case_Control) %>% 
  add_p() %>%
  add_n() %>% as_gt()
```

## Genes mutated in CHIP per patient

```{r}
muts_data %>% 
  distinct(NGS_ID, GENE, .keep_all = TRUE) %>% 
  filter(!is.na(GENE)) %>% group_by(Case_Control,GENE) %>% 
  summarise(count=n()) %>% 
  mutate(percent=(count/sum(count)*100)
  )

muts_data %>% 
  distinct(NGS_ID, GENE, .keep_all = TRUE) %>% 
  filter(!is.na(GENE)) %>% group_by(Case_Control,GENE) %>% 
  summarise(count=n()) %>% 
  mutate(percent=(count/sum(count)*100)
  ) %>% 
  ggplot(aes(x=reorder(GENE, percent), y=percent, fill = Case_Control)) +
  geom_bar(stat = "identity", position = position_dodge2(preserve = "single")) + 
  coord_flip()

muts_data %>% filter(!is.na(Case_Control)) %>% 
  distinct(NGS_ID, GENE, .keep_all = TRUE) %>% 
  select(Case_Control, GENE) %>% 
  tbl_summary(by=Case_Control) %>% 
  add_p() %>%
  add_n() %>% as_gt()
```

# Clinical mining
## Cancer repartition pie
```{r}
tbl <- as.data.frame(table(global_data$CANCER))
colourCount = length(unique(tbl$Var1))
getPalette = colorRampPalette(RColorBrewer::brewer.pal(8, "Accent"))(colourCount)
tbl %>% mutate(Var1 = fct_reorder(Var1, desc(Freq))) %>%
  ggplot(aes(x="", y=Freq, fill=Var1)) +
  geom_bar(stat="identity", width=1) +
  scale_fill_manual(name = "Cancer type", values = getPalette) +
  theme_minimal() +
  coord_polar("y", start=0, direction=-1) +
  labs(x="", y="", title="Cancer type Repartition")
```

## Number patient per Cancer
```{r}
tbl <- as.data.frame(table(global_data$CHIP,global_data$CANCER))
tbl %>% mutate(Var2 = fct_reorder(Var2, desc(Freq))) %>%
  ggplot(aes(x=Var2, y=Freq, fill=Var1)) +
  geom_bar(stat="identity", width = .75, position = position_dodge(preserve = "single")) +
  scale_fill_discrete(name = "CHIP") +
  geom_text(aes(y = Freq + .5 , label=Freq), position = position_dodge(width = 0.9)) +
  theme_minimal() +
  coord_flip()+
  labs(x="Cancer Type", y="Number of Patient", title="CHIP in Cancer type Repartition")

muts_data %>% group_by(CANCER,CHIP) %>% 
  summarise(count=n()) %>% 
  mutate(percent=(count/NROW(muts_data$CANCER)*100)
  ) %>% 
  ggplot(aes(x=reorder(CANCER, percent), y=percent, fill=CHIP)) +
  geom_bar(stat="identity", width = .75, position = position_dodge(preserve = "single")) +
  scale_fill_discrete(name = "CHIP") +
  theme_minimal() +
  coord_flip()
```

## Age per Cancer
```{r}
ggplot(data = global_data, aes(x=CANCER, y=Age), fill=CANCER) +
  geom_boxplot(color= magma(n=22)) +
  theme_minimal() +
  labs(x="Cancer type", y="Age", title="Age repartition per cancer") +
  coord_flip() +
  geom_jitter(shape=16, position=position_jitter(0.2))
```


## Gender
```{r}
ggplot(data = global_data, aes(x=Gender, y=Age), fill=Gender) +
  geom_boxplot() +
  theme_minimal() +
  labs(x="Gender", y="Age", title="Age repartition") +
  geom_jitter(shape=16, position=position_jitter(0.2)) +
  facet_grid(. ~ CHIP)
```

## Race
```{r}
ggplot(data = global_data, aes(x=Race, y=Age), fill=Race) +
  geom_boxplot() +
  theme_minimal() +
  labs(x="Race", y="Age", title="Age repartition") +
  geom_jitter(shape=16, position=position_jitter(0.2)) +
  facet_grid(. ~ CHIP)
```

## Ethnicity
```{r}
ggplot(data = global_data, aes(x=Ethnicity, y=Age), fill=Ethnicity) +
  geom_boxplot() +
  theme_minimal() +
  labs(x="Ethnicity", y="Age", title="Age repartition") +
  geom_jitter(shape=16, position=position_jitter(0.2)) +
  facet_grid(. ~ CHIP)
```

## Metastasis
```{r}
ggplot(data = global_data, aes(x=Mets, y=Age), fill=Mets) +
  geom_boxplot() +
  theme_minimal() +
  labs(x="Mets", y="Age", title="Age repartition") +
  geom_jitter(shape=16, position=position_jitter(0.2)) +
  facet_grid(. ~ CHIP)
```

## Neutro, Anemia, Thrombo, Leuko in CHIP
```{r}
global_data %>% 
  select(CHIP,
         Neutro, Anemia, Thrombo, Leuko
  ) %>% 
  tbl_summary(by= CHIP, statistic = all_continuous() ~ "{median} ({sd})") %>% 
  add_p() %>%
  add_n()
```

## Table 1_Patient Population----
### Case_Control
```{r}
global_data %>% 
  select(Case_Control, Age, Gender, Race, Ethnicity, Smoking, Mets, 
         BaseANC, BaseHGB, BasePLT, BaseWBC,
         ChangeANC, ChangeHGB, ChangePLT, ChangeWBC, 
         Prior_chemo, Prior_rad, 
         MAX2, MAX2heme
  ) %>% 
  tbl_summary(by= Case_Control, statistic = all_continuous() ~ "{median} ({sd})") %>% 
  add_p() %>%
  add_n() %>% as_gt()
```

### CHIP
```{r}
global_data %>% 
  select(CHIP, Case_Control, Age, Gender, Race, Ethnicity, Smoking, Mets, 
         BaseANC, BaseHGB, BasePLT, BaseWBC,
         ChangeANC, ChangeHGB, ChangePLT, ChangeWBC, 
         Prior_chemo, Prior_rad, 
         MAX2, MAX2heme
  ) %>% 
  tbl_summary(by= CHIP, statistic = all_continuous() ~ "{median} ({sd})") %>% 
  add_p() %>%
  add_n() %>% as_gt()
```

### Case_Control and CHIP
```{r}
tbl1 <- global_data %>% 
  filter(CHIP == "CHIP") %>% 
  select(Case_Control, Age, Gender, Race, Ethnicity, Smoking, Mets, 
         BaseANC, BaseHGB, BasePLT, BaseWBC,
         ChangeANC, ChangeHGB, ChangePLT, ChangeWBC, 
         Prior_chemo, Prior_rad, 
         MAX2, MAX2heme
  ) %>% 
  tbl_summary(by= Case_Control, statistic = all_continuous() ~ "{median} ({sd})") %>% 
  add_p() %>%
  add_n()
tbl2 <- global_data %>% 
  filter(CHIP == "No CHIP") %>% 
  select(Case_Control, Age, Gender, Race, Ethnicity, Smoking, Mets, 
         BaseANC, BaseHGB, BasePLT, BaseWBC,
         ChangeANC, ChangeHGB, ChangePLT, ChangeWBC, 
         Prior_chemo, Prior_rad, 
         MAX2, MAX2heme
  ) %>% 
  tbl_summary(by= Case_Control, statistic = all_continuous() ~ "{median} ({sd})") %>% 
  add_p() %>%
  add_n()
tbl_merge(list(tbl1, tbl2),
                 tab_spanner = c("**CHIP**", "**no CHIP**"))  %>% as_gt()
```

# Myelosuppression
## What is the most correlated with myelosupp?
### Neutro, Anemia, Thrombo, Leuko?

```{r, echo=TRUE}
summary(table(global_data$Case_Control, global_data$Neutro))
summary(table(global_data$Case_Control, global_data$Anemia))
summary(table(global_data$Case_Control, global_data$Thrombo))
summary(table(global_data$Case_Control, global_data$Leuko))
```
```{r}
global_data %>% 
  select(Case_Control,
         Neutro, Anemia, Thrombo, Leuko
  ) %>% 
  tbl_summary(by= Case_Control, statistic = all_continuous() ~ "{median} ({sd})") %>% 
  add_p() %>%
  add_n()
```
```{r, echo=TRUE}
logistic <- glm(C_C ~ Neutro+Anemia+Thrombo+Leuko,
                data = global_data, family = "binomial")
summary(logistic)
```
```{r, echo=TRUE}
logistic <- glm(C_C ~ Neutro+Thrombo+Leuko,
                data = global_data, family = "binomial")
summary(logistic)
```

### CBC?
```{r, echo=TRUE}
logistic <- glm(C_C ~ CHIP+BaseANC+BaseHGB+BasePLT+BaseWBC+ChangeANC+ChangeHGB+ChangePLT+ChangeWBC+MAX2heme,
                data = global_data, family = "binomial")
summary(logistic)
```
```{r, echo=TRUE}
logistic <- glm(C_C ~ BaseANC+BaseHGB+BasePLT+BaseWBC+ChangeANC+ChangeHGB+ChangePLT+ChangeWBC,
                data = global_data, family = "binomial")
summary(logistic)
```
```{r, echo=TRUE}
logistic <- glm(C_C ~ BaseANC+ChangeANC,
                data = global_data, family = "binomial")
summary(logistic)
```
```{r, echo=TRUE}
logistic <- glm(C_C ~ CHIP+BaseANC+ChangeANC,
                data = global_data, family = "binomial")
summary(logistic)
```
### But is it better if separate CHIP vs noCHIP?
```{r, fig.width = 11, fig.height = 8}
p1 <- global_data %>% gather("blood_subset", "value", c("BaseANC", "ChangeANC",
                                                  "BaseHGB", "ChangeHGB",
                                                  "BasePLT", "ChangePLT",
                                                  "BaseWBC", "ChangeWBC")) %>% 
  select("NGS_ID", "blood_subset", "value", "CHIP", "Case_Control") %>% 
  mutate(blood_subset= factor(blood_subset, levels = c("BaseANC", "ChangeANC",
                                                       "BaseHGB", "ChangeHGB",
                                                       "BasePLT", "ChangePLT",
                                                       "BaseWBC", "ChangeWBC"))) %>% 
  ggplot(aes(x=Case_Control, y=value))+
  geom_boxplot()+
  facet_wrap(. ~ blood_subset, scales = "free",  ncol=2, strip.position = "right")+
  stat_compare_means(size= 3)
df2 <- data.frame(CHIP = rep("CHIP",4), blood_subset = factor(c("BaseANC", "ChangeANC",
                                                                "BaseHGB", "ChangeHGB",
                                                                "BasePLT", "ChangePLT",
                                                                "BaseWBC", "ChangeWBC")),
                  value = c(10,10,16,NA, NA, NA, 10,10), Case_Control = rep("Cases",4))
p2 <- global_data %>% gather("blood_subset", "value", c("BaseANC", "ChangeANC",
                                                  "BaseHGB", "ChangeHGB",
                                                  "BasePLT", "ChangePLT",
                                                  "BaseWBC", "ChangeWBC")) %>% 
  select("NGS_ID", "blood_subset", "value", "CHIP", "Case_Control") %>% 
  mutate(blood_subset= factor(blood_subset, levels = c("BaseANC", "ChangeANC",
                                                       "BaseHGB", "ChangeHGB",
                                                       "BasePLT", "ChangePLT",
                                                       "BaseWBC", "ChangeWBC"))) %>% 
  ggplot(aes(x=CHIP, y=value, fill=Case_Control))+
  geom_boxplot()+
  geom_point(data = df2, aes(x = CHIP, y = value, fill=Case_Control), colour = "white", alpha=0) +
  facet_wrap(. ~ blood_subset, scales = "free",  ncol=2, strip.position = "right")+
  stat_compare_means(size= 3)
gridExtra::grid.arrange(p1, p2, ncol = 2, widths= c(4,6))
```

```{r, fig.width = 8, fig.height = 8}
global_data %>% gather("blood_subset", "value", c("BaseANC", "ChangeANC",
                                                  "BaseHGB", "ChangeHGB",
                                                  "BasePLT", "ChangePLT",
                                                  "BaseWBC", "ChangeWBC")) %>% 
  select("NGS_ID", "blood_subset", "value", "CHIP", "Case_Control") %>% 
  mutate(blood_subset= factor(blood_subset, levels = c("BaseANC", "ChangeANC",
                                                       "BaseHGB", "ChangeHGB",
                                                       "BasePLT", "ChangePLT",
                                                       "BaseWBC", "ChangeWBC"))) %>% 
  ggplot(aes(x=Case_Control, y=value, fill=CHIP))+
  geom_boxplot()+
  facet_wrap(. ~ blood_subset, scales = "free",  ncol=2, strip.position = "right")+
  stat_compare_means(size= 3)
```








